src/components/Overlay.tsx

```diff
@@ -12,22 +12,29 @@ const Overlay = ({
   onClose: () => void
   children: ReactNode
 }) => {
+  const dialogRef = useRef<HTMLDialogElement>(null)
+
+  useEffect(() => {
+    if (isOpen) {
+      dialogRef.current?.showModal()
+    } else {
+      dialogRef.current?.close()
+    }
+  })
   return (
-    <div
-      className={`fixed inset-0  flex items-center justify-center p-10 ${
-        isOpen ? 'block' : 'hidden'
-      } z-50 bg-gray-950/[.80] `}
+    <dialog
+      ref={dialogRef}
```

Comment:
Can you add a `fixed` class to this to make sure it stays in position if the user scrolls?